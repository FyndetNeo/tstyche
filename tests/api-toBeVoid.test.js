import { expect, test } from "@jest/globals";
import * as tstyche from "tstyche";
import { getFixtureUrl } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixtureUrl = getFixtureUrl("api-toBeVoid");

test("'toBeVoid' implementation", () => {
  expect(tstyche.expect).toHaveProperty("type.toBeVoid", expect.any(Function));
});

test("toBeVoid", async () => {
  const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

  expect(exitCode).toBe(1);
});
