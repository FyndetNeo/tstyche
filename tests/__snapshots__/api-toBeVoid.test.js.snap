// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeVoid: stderr 1`] = `
"Error: Type 'void' is not identical to type 'string'.

   7 |   expect(returnsVoid()).type.toBeVoid();
   8 | 
>  9 |   expect(returnsString()).type.toBeVoid();
     |                                ^
  10 | });
  11 | 
  12 | test("is NOT void?", () => {

       at ./__typetests__/toBeVoid.tst.ts:9:32 ❭ is void?

Error: Type 'void' is identical to type 'void'.

  13 |   expect(returnsString()).type.not.toBeVoid();
  14 | 
> 15 |   expect(returnsVoid()).type.not.toBeVoid();
     |                                  ^
  16 | });
  17 | 

       at ./__typetests__/toBeVoid.tst.ts:15:34 ❭ is NOT void?

"
`;

exports[`toBeVoid: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/toBeVoid.tst.ts
  × is void?
  × is NOT void?

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      2 failed, 2 total
Assertions: 2 failed, 2 passed, 4 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
